src/core/classes/site.ts

```diff
@@ -111,7 +115,7 @@ export class CoreSite {
     // Rest of variables.
     protected logger: CoreLogger;
     protected db?: SQLiteDB;
-    protected cacheTable: AsyncInstance<CoreDatabaseTable<CoreSiteWSCacheRecord>>;
+    protected cacheTables: Record<string, AsyncInstance<CoreDatabaseTable<CoreSiteWSCacheRecord>>> = {};
```

Comment:
```suggestion
    protected cacheTables: Record<WSCacheTablesSuffixes, AsyncInstance<CoreDatabaseTable<CoreSiteWSCacheRecord>>> = {};
```
I don't think in practice this will have any relevant impact, because we're initializing the tables using the enum already. But I think it's good practice to do this whenever possible.

If you accept this change, you won't be able to initialize the object like this (which to be honest, is good because we don't want an empty object to be a valid value for this property). But since you're already initializing the object in the constructor, it should be fine. It's just that instead of populating the object with a forEach, you'll have to implement it with a reduce and assign the variable.

I'm thinking you may have some issues declaring types for the reduce though, because until all the values have been added it won't be complete. I guess we could do this "the right way" using (or implementing) some helpers to map enums into values or something, but I think it's ok if you just cast it manually.

Reply:
Changed!